### ISSUE TITLE
Migrate to VNC

### ISSUE THREAD

**Issue by ouziel-slama** [towercomputers/toweros#95]
* Moonlight/Sunshine — https://news.ycombinator.com/item?id=37688887

**Comment by adamkrellenstein** [comment]
Is this fully OSS? More complicated to implement?

What are the downsides??

**Comment by ouziel-slama** [comment]
Well I tested the 3 alternatives:

**waypipe**

Easy to install and use. Fast and good quality. Not secure enough. Sound ?. 

Question: is it secure enough for tower-cli to support waypipe in addition to nx, as an option? @adamkrellenstein 

**wayvnc**

Easy to install and use (less than waypipe however).
Not very fast (there are two cursors, the host's cursor follows the client's with a little lag)
Really mediocre quality (we can perhaps improve but I haven't found any obvious adjustments)
I haven't found a way to launch a single application. Wayvnc shares the entire desktop (sway because labwc does not work with wayvnc). We could possibly launch the application in full screen and hide the desktop inside the host but we must first resolve the quality problem. (For the moment to have correct quality with `foo` I have to put the vncviewer in fullscreen)

**Moolight/Sunshine**

No apks and very complicated to compile (musl and arm don't make it easy). There are tons of dependencies to compile, so far I've spent an hour or two on it and I haven't managed to compile the first one. I think we should test it on a compatible distribution (Ubuntu, Debian, Fedora) and be sure that it does the job before starting on this project ((especially can we run a single application headless or is the entire desktop/screen shared like with wayvnc).


IMHO, for the moment `nx` remains the best option if we look overall at simplicity/quality/security. The big problem with `nx` at the moment is the startup time (and the lack of docs but this obstacle is already almost overcome) and we could solve it with a small loading dialog box.
